homeassistant/components/automation/numeric_state.py

```diff
@@ -81,15 +79,17 @@ def call_action():
 
         matching = check_numeric_state(entity, from_s, to_s)
 
-        if matching and not already_triggered:
+        if matching and entity not in entities_triggered:
             if time_delta:
-                async_remove_track_same = async_track_same_state(
+                unsub_track_same[entity] = async_track_same_state(
                     hass, time_delta, call_action, entity_ids=entity_id,
                     async_check_same_func=check_numeric_state)
             else:
                 call_action()
 
-        already_triggered = matching
+            entities_triggered.add(entity)
+        else:
+            entities_triggered.discard(entity)
```

Comment:
This discards the entity on the second match, thus making the trigger fire again on the third match even though the threshold was never crossed (right?)

Reply:
fixed thanks